state/operation.go

```diff
@@ -191,29 +196,30 @@ func newOperation(st *State, doc operationDoc, taskStatus []ActionStatus) Operat
 }
 
 // newOperationDoc builds a new operationDoc.
-func newOperationDoc(mb modelBackend, summary string) (operationDoc, string, error) {
+func newOperationDoc(mb modelBackend, summary string, count int) (operationDoc, string, error) {
 	id, err := sequenceWithMin(mb, "task", 1)
 	if err != nil {
 		return operationDoc{}, "", errors.Trace(err)
 	}
 	operationID := strconv.Itoa(id)
 	modelUUID := mb.modelUUID()
 	return operationDoc{
-		DocId:     mb.docID(operationID),
-		ModelUUID: modelUUID,
-		Enqueued:  mb.nowToTheSecond(),
-		Status:    ActionPending,
-		Summary:   summary,
+		DocId:             mb.docID(operationID),
+		ModelUUID:         modelUUID,
+		Enqueued:          mb.nowToTheSecond(),
+		Status:            ActionPending,
+		Summary:           summary,
+		ExpectedTaskCount: count,
```

Comment:
Just in case there is a long-running operation in progress when upgrading to a version with this fix, should we also perhaps include an upgrade step to populate this field for existing (maybe just the not yet completed ones) operations?

Reply:
Pondering how that will work.  It's entirely possible that all tasks have not been added before the upgrade.  In that case, the SpawedTaskCount would not be accurate.

This reminded me that I need a migration step.